### DISCUSSION TITLE
Custom edge

### DISCUSSION THREAD

**Discussion by Aman-singhmar** [xyflow/xyflow#3972] [Q&A]
I want to add a custom edge with a button.
But I need a label and button on the custom edge with type: 'smoothstep' style. 
Is it possible?

**Comment by bcakmakoglu** [comment]
Did you take the time to go through the [`Custom Edges`](https://reactflow.dev/examples/edges/custom-edges) example?
The example shows you how to do an edge with a button.
To use a smoothstep edge instead of a bezier, simply use the [`getSmoothStepPath`](https://reactflow.dev/api-reference/utils/get-smooth-step-path) function instead of `getBezierPath`.
